src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -1224,83 +1224,144 @@ CATCH_RETURN();
 #pragma endregion
 
 #pragma region internal methods for mimicking text analyzer pattern but for font fallback
+// Method Description:
+// - Converts a DWRITE_FONT_STRETCH enum into the corresponding float value to
+//   create a DWRITE_FONT_AXIS_VALUE with
+// Arguments:
+// - fontStretch: the old DWRITE_FONT_STRETCH enum to be converted into an axis value
+// Return value:
+// - The float value corresponding to the passed in fontStretch
+float CustomTextLayout::_FontStretchToWidthAxisValue(const DWRITE_FONT_STRETCH fontStretch)
+{
+    switch (fontStretch)
+    {
+    case DWRITE_FONT_STRETCH_ULTRA_CONDENSED:
+        return 50.0f;
+    case DWRITE_FONT_STRETCH_EXTRA_CONDENSED:
+        return 62.5f;
+    case DWRITE_FONT_STRETCH_CONDENSED:
+        return 75.0f;
+    case DWRITE_FONT_STRETCH_SEMI_CONDENSED:
+        return 87.5f;
+    case DWRITE_FONT_STRETCH_MEDIUM:
+        return 100.0f;
+    case DWRITE_FONT_STRETCH_SEMI_EXPANDED:
+        return 112.5f;
+    case DWRITE_FONT_STRETCH_EXPANDED:
+        return 125.0f;
+    case DWRITE_FONT_STRETCH_EXTRA_EXPANDED:
+        return 150.0f;
+    case DWRITE_FONT_STRETCH_ULTRA_EXPANDED:
+        return 200.0f;
+    default:
+        return 100.0f;

```

Comment:
Hey look @lhecker.... your favorite!

Reply:
FYI @PankajBhojwani With gcc or clang this wouldn't be an issue at all - in fact it would be the absolutely perfect, readable code, but unfortunately MSVC does a very poor job at optimizing switch/cases in general.
You can see the drastic difference here: https://godbolt.org/z/5YnTa3jPa

That said this code would be optimal:
```cpp
float CustomTextLayout::_FontStretchToWidthAxisValue(DWRITE_FONT_STRETCH fontStretch)
{
    // 10 elements from DWRITE_FONT_STRETCH_UNDEFINED (0) to DWRITE_FONT_STRETCH_ULTRA_EXPANDED (9).
    static constexpr auto data = std::array{100.0f, 50.0f, 62.5f, 75.0f, 87.5f, 100.0f, 112.5f, 125.0f, 150.0f, 200.0f};

    if (fontStretch > data.size()) {
        fontStretch = 0;
    }

    return data[fontStretch];
}
```

While it's now less obvious now which array element belongs to which `DWRITE_FONT_STRETCH` enum value, I doubt that a whole lot of people are going to re-read this particular code in the future. And if they do the comment should help. 🙂

Reply:
Wonder if we want to move this & similar methods inside `DxFontInfoData`, which is meant to be where the DX font things are.

Reply:
> Wonder if we want to move this & similar methods inside DxFontInfoData, which is meant to be where the DX font things are.

I'm guessing you mean `DxFontRenderData` right?

Reply:
Ah yeah. That’s what happen when you have many children: You don’t remember their names 🙃